### ISSUE TITLE
Implement Age 3 cards for Artifacts Expansion

### ISSUE THREAD

**Issue by ultimatefiend** [micahstairs/bga-innovation#97]
This issue will track the implementation of all Age 3 cards for the Artifacts expansion. Please update this comment before working on any of these cards (if the work is unblocked) so that we don't have duplicate work being done. We no longer need to file a separate issue for each card, since a consolidated list of the cards will keep us more organized.

  - [x] **135 - Dunhuang Star Chart**
  - [x] **136 - Charter of Liberties**
  - [x] **137 - Excalibur**
  - [x] **138 - Mjolnir Amulet**
  - [x] **139 - Philosopher's Stone**
  - [x] **140 - Beauvais Cathedral Clock**
  - [x] **141 - Along the River During the Qingming Festival**
  - [x] **142 - Moylough Belt Shrine**
  - [x] **143 - Necronomicon**
  - [x] **144 -Shroud of Turin**

**Comment by ultimatefiend** [comment]
Took another look at Charter of Liberties.  This one will be a bear.

The "select any splayed pile on anyone's board" is going to require a very specific interaction.  In a four person game, you would need to iterate through everyone's board to see which piles are eligible.  I guess you can do a 'choose_color' for each player.  Anyway, I'm not sure how to do that....

**Comment by micahstairs** [comment]
> Took another look at Charter of Liberties. This one will be a bear.
> 
> The "select any splayed pile on anyone's board" is going to require a very specific interaction. In a four person game, you would need to iterate through everyone's board to see which piles are eligible. I guess you can do a 'choose_color' for each player. Anyway, I'm not sure how to do that....

So one option that comes to mind is that I could add an `'has_splay_direction' => array(1, 2, 3)` option which would only select cards if they are splayed left, right, or up. It's generic enough that we do a lot with it later if we need to. I'm quite familiar with adding this sort of functionality so I could do it quite quickly. Feel free to file an issue for me!

**Comment by ultimatefiend** [comment]
sure, that should help.  I was thinking this could be done in two stages : One would select the player's board and the second would select

A way to select a player's board would also be useful, like a "choose_player" choice which could be restricted based on what the options array needs.  I am looking at communist manifesto (age 7 purple) and that would certainly benefit.

**Comment by micahstairs** [comment]
Hmm Communist Manifesto will indeed be a tricky onel!

**Comment by micahstairs** [comment]
For Charter of Liberties, I'd prefer not to make it a 2-step process, since they we'd probably need a way for them to undo their first selection. It's harder to make a mistake if they can directly pick the player and color at the same time.
